pkg/controllers/disruption/suite_test.go

```diff
@@ -467,6 +467,105 @@ var _ = Describe("Pod Eviction Cost", func() {
 	})
 })
 
+var _ = Describe("Nomination", func() {
+	var nodePool *v1beta1.NodePool
+	var nodeClaims []*v1beta1.NodeClaim
+	var nodes []*v1.Node
+	It("should nominate pods at the end of the disruption loop to existing nodes", func() {
+		nodePool = test.NodePool()
+		nodePool.Spec.Disruption.Budgets = []v1beta1.Budget{{
+			Nodes: "100%",
+		}}
+
+		// create our RS so we can link a pod to it
+		rs := test.ReplicaSet()
+		ExpectApplied(ctx, env.Client, rs)
+		Expect(env.Client.Get(ctx, client.ObjectKeyFromObject(rs), rs)).To(Succeed())
+
+		// Create a two pods that can all fit on one node.
+		pods := test.Pods(2, test.PodOptions{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{"app": "large-app"},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion:         "apps/v1",
+						Kind:               "ReplicaSet",
+						Name:               rs.Name,
+						UID:                rs.UID,
+						Controller:         ptr.Bool(true),
+						BlockOwnerDeletion: ptr.Bool(true),
+					},
+				},
+			},
+			// Each 2xlarge has 8 cpu, so each node should fit 2 pods.
+			ResourceRequirements: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceCPU: resource.MustParse("3"),
+				},
+			},
+		})
+
+		nodeClaims, nodes = test.NodeClaimsAndNodes(2, v1beta1.NodeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					v1beta1.NodePoolLabelKey:     nodePool.Name,
+					v1.LabelInstanceTypeStable:   mostExpensiveInstance.Name,
+					v1beta1.CapacityTypeLabelKey: mostExpensiveOffering.CapacityType,
+					v1.LabelTopologyZone:         mostExpensiveOffering.Zone,
+				},
+			},
+			Status: v1beta1.NodeClaimStatus{
+				Allocatable: map[v1.ResourceName]resource.Quantity{
+					v1.ResourceCPU:  resource.MustParse("8"),
+					v1.ResourcePods: resource.MustParse("100"),
+				},
+			},
+		})
+		nodeClaims[0].StatusConditions().MarkTrue(v1beta1.Drifted)
+		nodeClaims[1].StatusConditions().MarkTrue(v1beta1.Drifted)
+
+		ExpectApplied(ctx, env.Client, nodePool, pods[0], nodes[0], nodeClaims[0])
+		ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*v1.Node{nodes[0]}, []*v1beta1.NodeClaim{nodeClaims[0]})
+
+		// Expect the first pod scheduled, and then cordon the node so that the other pod can schedule
```

Comment:
Why do you need to cordon at all if consolidation should just take care of combining the nodes -- either A onto B or B onto A

Reply:
You bring up a good point